packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -197,16 +192,22 @@ export class TreeNodeKernel {
 
 	/**
 	 * Transition from {@link Unhydrated} to hydrated.
-	 * Bi-directionally associates the given hydrated TreeNode to the given anchor node.
+	 * Bi-directionally associates the given hydrated TreeNode to the anchor node at the provided path.
 	 * @remarks
 	 * Happens at most once for any given node.
-	 * Cleans up mappings to {@link UnhydratedFlexTreeNode} - it is assumed that they are no longer needed once the proxy has an anchor node.
+	 * Cleans up mappings to {@link UnhydratedFlexTreeNode} - it is assumed that they are no longer needed once this node has an anchor node.
 	 */
-	private hydrate(anchorNode: AnchorNode): void {
+	public hydrate(anchors: AnchorSet, path: UpPath): void {
```

Comment:
What does this do if called more than once? Is that permitted?

Reply:
It will assert (see note https://github.com/microsoft/FluidFramework/pull/24626#discussion_r2089695262 ). Looking at the tests for this file though, they seem a bit lacking. I have now added test coverage for this and a few other things.